components/style/values/generics/mod.rs

```diff
@@ -144,3 +145,16 @@ pub struct NonNegative<T>(pub T);
 #[derive(Animate, Clone, ComputeSquaredDistance, Copy, Debug, MallocSizeOf)]
 #[derive(PartialEq, PartialOrd, ToAnimatedZero, ToComputedValue, ToCss)]
 pub struct GreaterThanOrEqualToOne<T>(pub T);
+
+/// Common handling for generic CSS url() values.
+pub mod url {
+    /// An image url or none, used for example in list-style-image
+    #[derive(Animate, Clone, ComputeSquaredDistance, Debug, MallocSizeOf, PartialEq)]
+    #[derive(ToAnimatedValue, ToComputedValue, ToCss)]
+    pub enum ImageUrlOrNone<Url> {
```

Comment:
Also, this looks more like `UrlOrNone` now, wdyt? This doesn't have anything specific about images.

You can also implement `Parse` as follows on this type:

```rust
impl<Url: Parse> Parse for UrlOrNone<Url> {
    // ...
}
```

That means that the current `UrlOrNone` can also be replaced by this type, making this something like:

## Specified

```rust
type UrlOrNone = UrlOrNone<SpecifiedUrl>;
type ImageUrlOrNone = UrlOrNone<SpecifiedImageUrl>;
```

## Computed

```rust
type UrlOrNone = UrlOrNone<ComputedUrl>;
type ImageUrlOrNone = UrlOrNone<ComputedImageUrl>;
```

And I think we can also get rid of the `ListStyleImage` type completely, and just use `ImageUrlOrNone`.

WDYT? It'd be cleaner I think, though it's a fair change, so we can also leave it for a followup maybe? I'd prefer to do it in this PR directly but...